File: 03-Sliding-Window/16-Longest-Substring-Without-Repeating-Characters/lengthOfLongestSubstring.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function (s) {
  if (!s) {
    return 0;
  }
  let l = 0;
  let r = 1;
  let maxLen = 1;

  while (r < s.length) {
    if (s.substring(l, r).indexOf(s[r]) === -1) {
      maxLen = Math.max(maxLen, r - l + 1);
    } else {
      let index = s.indexOf(s[r], l);
      if (index !== -1) {
        l = index + 1;
      }
    }
    r++;
  }
  return maxLen;
};

File: 03-Sliding-Window/16-Longest-Substring-Without-Repeating-Characters/lengthOfLongestSubstring.py
from typing import List


class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if not s:
            return 0
        l, r = 0, 1
        maxLen = 1
        while r < len(s):
            if s[r] not in s[l:r]:
                maxLen = max(maxLen, r-l+1)
            else:
                l = s.index(s[r], l, r)+1
            r += 1
        return maxLen
